api/src/main/java/io/opentelemetry/common/ReadableAttributes.java

```diff
@@ -21,4 +21,6 @@
  *
  * <p>See {@link Attributes} for the public API implementation.
  */
-public interface ReadableAttributes extends ReadableKeyValuePairs<AttributeValue> {}
+public interface ReadableAttributes extends ReadableKeyValuePairs<String, AttributeValue> {
+  AttributeValue getValue(String key);
```

Comment:
I don't quite follow the PR description about clashing with `Map` - we don't inherit from `Map` so there's no clash right?

Reply:
The issue is with the AttributeMap used in the RecordEventsReadableSpan. It does extend Map, and the `get` method on Map clashes with what this has now.

Reply:
I updated the PR description with some details that I realized was missing for people other than me. :)

Reply:
Maybe a better answer is to finally rip off the bandaid, and make the AttributesMap not extend HashMap.

Reply:
@anuraaga I made AttributesMap not extend HashMap anymore, and restored the `get` name. Please take a look!